serialization/include/rp_ser.h

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef RP_SER_H_
+#define RP_SER_H_
+
+#include <stdint.h>
+
+#include <cbor.h>
+
+#include <rp_errors.h>
+#include <rp_common.h>
+#include <rp_trans.h>
+
+/**
+ * @file
+ * @defgroup rp_ser Remote Procedures Serialization core
+ * @{
+ * @brief Remote Procedures Serialization core API.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**@brief Serialization packet type.*/
+enum rp_ser_packet_type {
+	/** Serialization command packet. */
+	RP_SER_PACKET_TYPE_CMD          = 0x01,
+
+	/** Serialization event packet. */
+	RP_SER_PACKET_TYPE_EVENT,
+
+	/** Serialization command response packet. */
+	RP_SER_PACKET_TYPE_RSP,
+
+	/** Serialization transport reserved packet. */
+	RP_SER_PACKET_TRANSPORT_RESERVED = 128,
+
+	/** Serialization upper bound. */
+	RP_SER_PACKER_TYPE_MAX           = 255
+};
+
+/**@brief Command and response decoder handler type. */
+typedef rp_err_t (*cmd_handler_t)(CborValue *it);
+
+/**@brief Event decoder handler type. */
+typedef rp_err_t (*evt_handler_t)(uint8_t evt, CborValue *it);
+
+/**@brief Encoder buffer structure. */
+struct rp_ser_buf {
+	/** Encoder buffer. */
+	uint8_t *buf;
+
+	/** Encoder buffer size. */
+	size_t size;
```

Comment:
Could you make it const or do you need to be able to change the buffer size within the same ```rp_ser_buf``` instance?